nx/blocks/snapshot-admin/views/snapshot.js

```diff
@@ -33,14 +36,21 @@ class NxSnapshot extends LitElement {
     _message: { state: true },
     _isOpen: { state: true },
     _action: { state: true },
+    _isRegisteredForSnapshotScheduler: { state: true, type: Boolean },
   };
 
   async connectedCallback() {
     super.connectedCallback();
+    this._isRegisteredForSnapshotScheduler = false;
     this.shadowRoot.adoptedStyleSheets = [style];
     getSvg({ parent: this.shadowRoot, paths: ICONS });
   }
 
+  async firstUpdated() {
+    // Check if registered for snapshot scheduler after first render
+    this._isRegisteredForSnapshotScheduler = await isRegisteredForSnapshotScheduler();
```

Comment:
If you're not waiting on DOM to render, you can do this in `connectedCallback` which should only run once.

`firstUpdated` is for use cases where you need to know the DOM inside the web component.

https://lit.dev/docs/components/lifecycle/#firstupdated (this whole page on lifecycle is great, btw)